.github/workflows/publish-docs.yml

```diff
@@ -0,0 +1,49 @@
+name: Publish Docs
+
+on:
+  push:
+    tags:
+      - 'docs-*'
+    branches:
+      - dev
+
+jobs:
+  publish-docs:
+    runs-on: ubuntu-latest
+    if: github.ref_type == 'tag' && startsWith(github.ref, 'refs/tags/docs-')
+    permissions:
+      contents: write
+    steps:
+      - uses: actions/checkout@v4
+      - uses: ./.github/actions/setup-project
+      - name: Setup Git
+        run: |
+          git config --global user.name 'github-actions[bot]'
+          git config --global user.email 'github-actions[bot]@users.noreply.github.com'
+      - name: Build and Deploy Docs
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
```

Comment:
The token is injected from github actions: 

* https://docs.github.com/en/actions/security-for-github-actions/security-guides/automatic-token-authentication#permissions-for-the-github_token

Not sure if this will work 100% its worth some iterating.